lgc/util/StartStopTimer.cpp

```diff
@@ -42,13 +43,17 @@ namespace {
 
 // =====================================================================================================================
 // Pass to start or stop a timer
-class StartStopTimer : public ModulePass {
+class StartStopTimer : public PassInfoMixin<StartStopTimer> {
 public:
-  static char ID;
-  StartStopTimer() : ModulePass(ID) {}
-  StartStopTimer(Timer *timer, bool starting) : ModulePass(ID), m_timer(timer), m_starting(starting) {}
+  StartStopTimer() {}
+  StartStopTimer(Timer *timer, bool starting) : m_timer(timer), m_starting(starting) {}
 
-  bool runOnModule(Module &module) override;
+  PreservedAnalyses run(Module &M, ModuleAnalysisManager &AM) {
+    runOnModule(M);
+    return PreservedAnalyses::all();
+  }
+
+  bool runOnModule(Module &module);
```

Comment:
Can you please rename it to runImpl to avoid confusion (runOnModule is strongly associated with the legacy PM)?